flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/catalog/hive/HiveCatalogITCase.java

```diff
@@ -157,37 +155,27 @@ public void testCsvTableViaAPI() throws Exception {
         tableEnv.registerCatalog("myhive", hiveCatalog);
         tableEnv.useCatalog("myhive");
 
-        TableSchema schema =
+        final TableSchema schema =
                 TableSchema.builder()
                         .field("name", DataTypes.STRING())
                         .field("age", DataTypes.INT())
                         .build();
 
-        FormatDescriptor format =
-                new OldCsv().field("name", Types.STRING()).field("age", Types.INT());
-
-        CatalogTable source =
-                new CatalogTableBuilder(
-                                new FileSystem()
-                                        .path(
-                                                this.getClass()
-                                                        .getResource("/csv/test.csv")
-                                                        .getPath()),
-                                schema)
-                        .withFormat(format)
-                        .inAppendMode()
-                        .withComment("Comment.")
-                        .build();
+        final Map<String, String> sourceOptions = new HashMap<>();
+        sourceOptions.put("connector.type", "filesystem");
+        sourceOptions.put("connector.path", getClass().getResource("/csv/test.csv").getPath());
+        sourceOptions.put("format.type", "csv");
+
+        CatalogTable source = new CatalogTableImpl(schema, sourceOptions, "Comment.");
```

Comment:
(I know this is deprecated, but it seems that the Hive implementation is still very much expecting this old stack and I didn't want to touch this here)

Reply:
Makes sense.